src/jbuf/jbuf.c

```diff
@@ -207,10 +205,13 @@ int jbuf_alloc(struct jbuf **jbp, uint32_t min, uint32_t max)
 	}
 
 out:
-	if (err)
+	if (err) {
 		mem_deref(jb);
-	else
+	}
+	else {
 		*jbp = jb;
+		mem_destructor(jb, jbuf_destructor);
```

Comment:
Needs to be set directly after mutex_alloc, before frame alloc.

Reply:
`jbuf_destructor()` crashes if `mutex_alloc()` was not successful.

Reply:
Right, but not after err check.

Reply:
```c
jb = mem_zalloc(sizeof(*jb), NULL);
... -no other allocations -
err = mutex_alloc(&jb->lock);
if (err)
   goto out;
mem_destructor(jb, jbuf_destructor);
```

Reply:
Okay, or the newest version with NULL pointer check? Which one you prefer?